File: lab/javascript/tecnicas/fila2.js
const FILA = [];


function is_empty(){
    return FILA.length == 0;
}


function enqueue(valor){
    FILA.push(valor);
}


function dequeue(){
    if (is_empty()){
        return null;
    }
    return FILA.pop(0);
}

function view(){
    console.log(FILA);
}

enqueue(10)
enqueue(20)
view()
let removido = dequeue()
console.log("Removido -", removido)
view()

File: lab/python/tecnicas/fila2.py
FILA = []


def is_empty():
    return len(FILA) == 0


def enqueue(valor):
    FILA.append(valor)


def dequeue():
    if is_empty():
        return None
    return FILA.pop(0)


def view():
    print(FILA)


if __name__ == "__main__":
    enqueue(10)
    enqueue(20)
    view()
    removido = dequeue()
    print("Removido -", removido)
    view()
